lib/core/include/stream_factory_utility.hpp

```diff
@@ -128,7 +128,61 @@ namespace irods::experimental::io
 
             return d;
         };
-    }
+    } // make_dstream_factory
+
+    /// Creates a factory that produces managed_dstream objects.
+    ///
+    /// \param[in] _conn_pool    \parblock
+    ///                          The connection pool serving connections for each stream object created
+    ///                          by the factory. The connection pool is not owned by the generated factory
+    ///                          and must out live any streams produced by the factory.
+    ///                          \endparblock
+    /// \param[in] _logical_path The full path to a data object. All streams will point to this data object.
+    /// \param[in] _replica_id   The replica number or resource name used to identify a specific replica.
+    ///
+    /// \tparam ReplicaIdType \parblock
+    ///                       The type of the information used to identify a specific replica. The following
+    ///                       types are supported:
+    ///                           - irods::experimental::io::replica_number
+    ///                           - irods::experimental::io::leaf_resource_name
+    ///                           - irods::experimental::io::root_resource_name
+    ///                       \endparblock
+    ///
+    /// \return A new factory function.
+    ///
+    /// \since 4.2.9
+    template <typename ReplicaIdType>
```

Comment:
I do not like this name. I'd like the name to tell the user that this type is for identifying a specific replica.

Perhaps:
- `ReplicaIdentifyingInfo` (eww)
- `TargetReplicaType`
- Something else?

Reply:
`ReplicaIdentifier`?

Reply:
I'll use that.

Reply:
done